contributors/devel/godep.md

```diff
@@ -1,127 +1,178 @@
 # Using godep to manage dependencies
 
 This document is intended to show a way for managing `vendor/` tree dependencies
-in Kubernetes. If you are not planning on managing `vendor` dependencies go here
-[Godep dependency management](development.md#godep-dependency-management).
+in Kubernetes. If you do not need to manage vendored dependencies, you probably
+do not need to read this.  Read 
+[Godep dependency management](development.md#godep-dependency-management)
+instead.
 
-## Alternate GOPATH for installing and using godep
+## Background
 
-There are many ways to build and host Go binaries. Here is one way to get
-utilities like `godep` installed:
+As a tool, `godep` leaves much to be desired.  It builds on `go get`, and adds
+the ability to pin dependencies to exact git version. The `go get` tool itself
+doesn't have any concept of versions, and tends to blow up if it finds a git
+repo synced to anything but `master`, but that is exactly the state that
```

Comment:
...and different versions of godep have preferred differing states! (not that you need to say this here)